### DISCUSSION TITLE
llama.cpp生成的量化模型，如何使用langchain进行推理

### DISCUSSION THREAD

**Discussion by Beyondliby27** [ymcui/Chinese-LLaMA-Alpaca-2#232] [Q&A]
用llama.cpp生成了ggml-model-q8_0.gguf这个模型权重文件，应该怎么样给langchain使用哇，请问有教程吗

**Comment by Beyondliby27** [comment]
只生成了gguf文件，没有生成对应的bin文件是为啥呀

**Comment by ymcui** [comment]
gguf就是之前的bin，扩展名换了而已

**Reply by Beyondliby27** [reply]
量化好模型之后，把对应配置的json也更改了，结果出现这个问题
OSError: Unable to load weights from pytorch checkpoint file for 'chinese-alpaca-2-7b-hf/ggml-model-q8_0.bin'

**Reply by Beyondliby27** [reply]
配置文件json里面的操作是，修改pytorch_model.bin.index里的model名字为，ggml-model-q8_0.bin

**Reply by Beyondliby27** [reply]
请问下，量化模型后，就是\quantize那个代码，是只输出一个gguf权重文件吗，我看没有json配置文件的改变

**Reply by ymcui** [reply]
我不是很明白你说的，你只需要按照我们wiki里的步骤来就可以了。
`./quantize`程序只会生成一个单独的模型文件，这个直接可以被llama.cpp读取，不需要什么json配置文件。

**Reply by Beyondliby27** [reply]
对，就是想问问量化后的模型能使用其他的推理方法吗？
量化的步骤是没有问题的，但是想用langchain去推理量化后的模型会报错，这种可以解决吗

**Reply by ymcui** [reply]
llama.cpp最新的GGUF模型格式和之前的不兼容。所以langchain能否支持最新版的GGUF格式模型得看langchain及其依赖是否已经做好适配了。这个建议你去langchain那边多做一些了解。
